src/services/Api/newsletters.ts

```diff
@@ -0,0 +1,127 @@
+import url from 'url';
+import { API_URL } from './base';
+
+export async function createNewsletter(
+  token: string,
+  newsletter: DraftNewsletter,
+  isTest: boolean,
+  pageCount: number,
+): Promise<NewsletterLog> {
+  let tagIds: number[] = [];
+  newsletter.tags.forEach((tag) => tagIds.push(tag.id));
+  const requestOptions = {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${token}`,
+    },
+    body: JSON.stringify({
+      name: newsletter.title,
+      s3_path: newsletter.file.name,
+      tags: tagIds,
+      is_test: isTest,
+      page_count: pageCount,
+    }),
+  };
+  const response = await fetch(
+    url.resolve(API_URL, 'newsletter'),
+    requestOptions,
+  );
+  const body = await response.json();
+  if (body.status === 'ERROR') {
+    throw body;
+  }
+  const newsletterData: NewsletterLog = {
+    id: body.data.id,
+    title: body.data.name,
+    fileLink: body.data.pdf_path,
+    delivered: body.data.delivered_count,
+    inTransit: body.data.in_transit_count,
+    returned: body.data.returned_count,
+    creationDate: new Date(body.data.created_at),
+    estimatedArrival: null,
+    tags: [],
+  };
+  if (body.data.estimated_arrival) {
+    newsletterData.estimatedArrival = new Date(body.data.estimated_arrival);
+  }
+  interface t {
+    id: number;
+    label: string;
+    total_contacts: number;
+  }
+  body.data.tags.forEach((tag: t) => {
+    const tagData: Tag = {
+      label: tag.label,
+      id: tag.id,
+      numContacts: tag.total_contacts,
+    };
+    newsletterData.tags.push(tagData);
+  });
+  return newsletterData;
+}
+
+export async function fetchNewsletters(
+  token: string,
+): Promise<NewsletterLog[]> {
+  const requestOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'text/plain',
+      Authorization: `Bearer ${token}`,
+    },
+  };
+  const response = await fetch(
+    url.resolve(API_URL, 'newsletters'),
+    requestOptions,
+  );
+  const body = await response.json();
+  if (body.status === 'ERROR') {
+    throw body;
+  }
+  interface t {
+    id: number;
+    label: string;
+    total_contacts: number;
+  }
+  interface n {
```

Comment:
same comment. I think we should move this to the Newsletter interface and make it extend a BaseNewsletter instead of duplicating the code to change it to pdf_path